### DISCUSSION TITLE
Keyboard adjustment with custom TextInput

### DISCUSSION THREAD

**Discussion by pavelglebov** [gorhom/react-native-bottom-sheet#1166] [General]
**Use case:**
1) I have bottomSheet with dynamic height (CONTENT_HEIGHT)
2) I have to use custom external TextInput (there is no way to implement same thing manually)

**Issue:**
BottomSheet height is not adjusted automatically when focusing custom TextInput.

Docs have following tip:
<img width="962" alt="Screen Shot 2022-11-02 at 18 21 03" src="https://user-images.githubusercontent.com/5727746/199558223-188bbca4-feb8-4a91-a4fe-b7144ebfa433.png">

When copy the code, including:
```
const { shouldHandleKeyboardEvents } = useBottomSheetInternal();
```

I get following error:
<img height="200" alt="Screen Shot 2022-11-02 at 18 21 03" src="https://user-images.githubusercontent.com/5727746/199558922-dd457d03-237d-48f2-a79f-3440a59fbd84.jpeg">

**Is there a way to trigger keyboard animation in bottomSheet from outside?**

**Comment by pavelglebov** [comment]
Okay, so the problem was that I was using useBottomSheetInternal hook from component that holds bottomSheet, not inside of it.
Moving it into the component that lays in the bottomSheet solved the issue.

**Reply by anhquan291** [reply]
Hi @pavelglebov,
I'm facing the same problem. Could you please share your custom textinput component? 
Really appreciate it! 

Thanks

**Reply by JocheAttah** [reply]
Please can you explain better? Hopefully with code

**Reply by lucas-hoffmann** [reply]
Could you please explain how you did this? I'm stuck with this error

**Reply by Karthik-B-06** [reply]
How did you make it work? Can you please share the code?

**Reply by Hoffmannn** [reply]
@Karthik-B-06 @JocheAttah 
I was able to solve the error. What happens is that you can't use `useBottomSheetInternal` in the same component as you use `<BottomSheetModal/>`.  

You have to create a separate component for the BottomSheet content and use the hook inside of it, passing this component as a child to `BottomSheetModal`. For example:

```javascript
<BottomSheetModal
   // your props as usual
  children={<BottomSheetContent/>}
 />
```
Then, outside of the component where you're using the BottomSheet (it can be in the same file, just not in the same exported component):
```javascript
const BottomSheetContent = ()=>{ 
    const { shouldHandleKeyboardEvents } = useBottomSheetInternal();
    return (
    {/* Your content here  */}
     );
};
```

**Reply by Karthik-B-06** [reply]
Great, thanks @Hoffmannn. This worked.

**Reply by theianjohnson** [reply]
@Hoffmannn @Karthik-B-06 @gorhom can you share how to copy/paste the onFocus/onBlur events? I have custom TextInputs that are also used outside a bottom sheet so I'm trying to figure out if I can pass the focus/blur callbacks into them or if I'd have to create a second bottom sheet specific TextInput (in which case I'd probably find another way to do this)

**Reply by Hoffmannn** [reply]
@theianjohnson I didn't have your exact situation, but I did implement a custom Input on my bottom sheet. I had to pass a function to handle keyboard events (move bottom sheet up when displaying the keyboard). For this I used the `useBottomSheetInternal` hook and simply passed this function to my onFocus/onBlur events:  
```javascript
 // BottomSheetContent
  const { shouldHandleKeyboardEvents } = useBottomSheetInternal();
  // ... 
  return (
     <View>
      <TextInputMask 
	onFocus={() => {
	shouldHandleKeyboardEvents.value = true;
	}}
	onBlur={() => {
	shouldHandleKeyboardEvents.value = false;
	}}
       />
        {/*  rest of the content*/}
      </View>
   )
```
